invenio_rdm_records/services/schemas/__init__.py

```diff
@@ -81,6 +84,19 @@ class Meta:
 
     #     return ExtensionSchema().load(value)
 
+    @validates("pids")
+    def validate_pids(self, value):
+        """Validates the keys of the pids are supported providers."""
+        # TODO, is this needed? Otherwise it will fail at sevice
+        # component level when processing. If needed add list of supported
+        # be careful on format not to explode the translation file
+
+        if set(value.keys()) - self.TMP_PIDS_TYPE:
```

Comment:
can you maybe use `==` instead?
```suggestion
        only_allowed_pids = set(value.keys()) == self.TMP_PIDS_TYPE
        if not only_allowed_pids:
```

Reply:
Discussed IRL, == does not cover all uses cases but I have extracted it to a variable to be more clear :)